.eslintrc.yml

```diff
@@ -196,7 +196,7 @@ rules:
   promise/no-return-wrap: error
   promise/param-names: off
   promise/prefer-await-to-callbacks: off
-  promise/prefer-await-to-then: off
+  promise/prefer-await-to-then: error
```

Comment:
I'm afraid I have to disagree with this one.
It creates more verbose code that is harder to read, especially in the executor.
Can we do optimization using just `then`?

Reply:
nope, then is broken

Reply:
but instead of using an immediately invoked anonymous arrow function, we could use arrow functions stored in variables. or even named functions, defined even on the top level; if the functions are defined on the top level, we would have to pass parameters rather than rely on the closure. I don't have a preference per se.

See [this comment](https://github.com/graphql/graphql-js/pull/3754#issue-1392076280) for links on how `then` is comparatively slower than async functions. I don't know how we can do optimizations and still use `then`. The entire point of #3754 and this PR is to avoid then as until it is fixed, it is too easy to unnecessarily add extra ticks.

Reply:
I settled on introducing new helpers `after`, `tryAfter`, `catchAfter` rather than using `.then` or immediately invoked anonymous arrow functions.

This reduces the diff considerably and I think addresses the readability concern.

We can also use these helpers to get rid of completePromisedValue, which I am about to do.

Reply:
Actually, probably better to do in a separate PR so that the diff is cleaner in `execute.ts`.

Reply:
Ready for re-review.